internal/rbd/controllerserver.go

```diff
@@ -743,9 +743,51 @@ func checkContentSource(
 	return nil, nil, status.Errorf(codes.InvalidArgument, "not a proper volume source")
 }
 
+// checkErrAndUndoReserve work on error from genVolFromVolID() and undo omap reserve.
+func (cs *ControllerServer) checkErrAndUndoReserve(
+	ctx context.Context,
+	err error,
+	volumeID string,
```

Comment:
is `volumeID` not part of the `rbdVol` that was passed?

Reply:
@nixpanic The volume id is getting filled from genVolFromVolID->generateVolumeFromVolumeID.. path , where the new function is operating on the error case of above call chain, we can not rely on the `rbdvol-rbdimage-voldID`, so the arg passing from the caller. hope it clarify.

Reply:
thanks for the explanation, please add that as a comment in the top of the function, otherwise someone might 'clean' it up at a later time and causing issues

Reply:
@nixpanic done. thanks. ptal